protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/stream/job/JobActivationPropertiesImpl.java

```diff
@@ -76,6 +76,12 @@ public List<String> getTenantIds() {
         .collect(Collectors.toList());
   }
 
+  public JobActivationPropertiesImpl setTenantIds(final List<StringValue> tenantIds) {
+    tenantIdsProp.reset();
+    tenantIds.forEach(tenantId -> tenantIdsProp.add().wrap(tenantId));
```

Comment:
Good idea, let's keep the `JobActivationPropertiesImpl` and `JobBatchRecord` behaviors aligned.

Instead of resetting it in `BrokerActivateJobsRequest.setTenantIds`, I'll create `JobBatchRecord#setTenantIds` where I'll reset and set the `tenantIdsProp`.